src/main/java/hu/oe/nik/szfmv/automatedcar/model/objects/Tree.java

```diff
@@ -0,0 +1,45 @@
+package hu.oe.nik.szfmv.automatedcar.model.objects;
+
+import java.awt.*;
+import java.awt.geom.AffineTransform;
+import java.awt.geom.Ellipse2D;
+
+public class Tree extends Stationary {
+
+    private static double D = 25;
+
+    /**
+     * Creates an object of the virtual world on the given coordinates with the given image.
+     *
+     *
+     * @param x             the initial x coordinate of the object
+     * @param y             the initial y coordinate of the object
+     * @param imageFileName the filename of the image representing the object in the virtual world
+     */
+    public Tree(int x, int y, String imageFileName) {
+        super(x, y, imageFileName);
+    }
+
+    /**
+     * Creates an object with default parameter values.
+     * Added default png
+     *
+     *
+     *
+     */
+    public Tree() {
+        super(0, 0, "tree.png");
+    }
+
+    @Override
+    public void generateShape() {
+        //get ellipse shape
+        AffineTransform t = new AffineTransform();
+        t.rotate(-this.getRotation(), this.getX(), this.getY());
```

Comment:
Erre a negatív rotation-ös dologra emlékszem. Gondolom kellőképpen le van dokumentálva és propagálva, hogy miért is van az ott.